### PR TITLE
Check alphabetization of CLI options

### LINKED ISSUE #524: Alphabetize command line options
Note: This issue is reserved for our new tech writer.

A few [command line options](https://docs.teku.consensys.io/reference/cli) are out of alphabetical order (such as `checkpoint-sync-url`). This issue is to put them back in order.

### PR DESCRIPTION
**Pull request by joaniefromtheblock**
Fixes #524 

- Reviewed CLI options and reordered entries that may be out of place

### CODE REVIEW

**Comment by joaniefromtheblock** [comment]
@alexandratran @mjsmike62 PTAL

File: docs/reference/cli/index.md [hunk 1]
```diff
@@ -1316,120 +1317,120 @@ metrics-block-timing-tracking-enabled: false
...
-
-:::tip
-
-To allow all hostnames, use `"*"`. We don't recommend allowing all hostnames for production environments.
-
-:::
+Set to `true` to enable the metrics exporter. The default is `false`.
 
-### metrics-categories
+### `metrics-host-allowlist
```
Comment (alexandratran) [inline docs/reference/cli/index.md:1392]:
```suggestion
### `metrics-host-allowlist`
```

**Review by alexandratran — COMMENTED** [review]
Just one missing backtick.

**Comment by joaniefromtheblock** [comment]
Thanks @alexandratran. Committed your suggestion :]

**Review by alexandratran — APPROVED** [review]
LGTM!

**Pull request merged**